src/ADAL.PCL.Desktop/CryptographyHelper.cs

```diff
@@ -54,33 +54,69 @@ public byte[] SignWithCertificate(string message, X509Certificate2 certificate)
         {
             if (certificate.PublicKey.Key.KeySize < ClientAssertionCertificate.MinKeySizeInBits)
             {
-                throw new ArgumentOutOfRangeException("rawData",
-                    string.Format(CultureInfo.InvariantCulture, AdalErrorMessage.CertificateKeySizeTooSmallTemplate, ClientAssertionCertificate.MinKeySizeInBits));
+                throw new ArgumentOutOfRangeException(nameof(certificate),
+                    string.Format(CultureInfo.InvariantCulture, AdalErrorMessage.CertificateKeySizeTooSmallTemplate,
+                        ClientAssertionCertificate.MinKeySizeInBits));
             }
 
-            X509AsymmetricSecurityKey x509Key = new X509AsymmetricSecurityKey(certificate);
-            RSACryptoServiceProvider rsa = x509Key.GetAsymmetricAlgorithm(SecurityAlgorithms.RsaSha256Signature, true) as RSACryptoServiceProvider;
+            byte[] messageBytes = Encoding.UTF8.GetBytes(message);
 
-            RSACryptoServiceProvider newRsa = null;
-            try
+            var x509Key = new X509AsymmetricSecurityKey(certificate);
+
+            using (RSA rsa = x509Key.GetAsymmetricAlgorithm(SecurityAlgorithms.RsaSha256Signature, true) as RSA)
```

Comment:
@jmprieur, I believe it will/does.
I shall try and repro it using MSAL and raise a bug if it does.